pyflink-walkthrough/README.md

```diff
@@ -102,6 +106,24 @@ $ docker-compose exec jobmanager ./bin/flink run -py /opt/pyflink-walkthrough/pa
 
 3. Stop the PyFlink job:
 
-Visit the Flink Web UI at [http://localhost:8081/#/overview](http://localhost:8081/#/overview) , select the job and click `Cancle` on the upper right side.
+Visit the Flink Web UI at [http://localhost:8081/#/overview](http://localhost:8081/#/overview) , select the job and click `Cancel` on the upper right side.
 
 ![image](pic/cancel.png)
+
+## Extension
+
+You are able to edit the [payment_msg_processing.py](payment_msg_proccessing.py) or create new PyFlink 
+projects to perform more complex processing logic locally on your operating system under the `pyflink-walkthrough` 
+directory since it is mounted on the `jobmanager` docker container. Such as:
+* Creating a new Kafka source table;
+* Creating a new index for the Elasticsearch sink;
+* Calculating the amount of transactions grouped by a 1 minute tumble window and payPlatforms.
+
+After the modification, you can submit the new job by executing the same command mentioned at 
+[Running the PyFlink Job](#running-the-pyflink-job)
+```shell script
+$ docker-compose exec jobmanager ./bin/flink run -py /opt/pyflink-walkthrough/payment_msg_proccessing.py -d
+```
+
+Furthermore, you can also [create new kibana dashboards](https://www.elastic.co/guide/en/kibana/7.8/dashboard-create-new-dashboard.html) 
+to visualize more charts of various dimension based on the persistent indexes in Elasticsearch.
```

Comment:
```suggestion
## Further Explorations

If you would like to explore this example more deeply, you can edit [payment_msg_processing.py](payment_msg_proccessing.py) or create new PyFlink 
projects that perform more complex processing. You can do this locally under the `pyflink-walkthrough` 
directory, since it is mounted on the `jobmanager` docker container. 

Ideas:
* Add your own Kafka source table;
* Create a new index for the Elasticsearch sink;
* Count the number of transactions, grouped by a 1 minute tumbling windows, and by payPlatform.

After making a modification, you can submit the new job by executing the same command mentioned at 
[Running the PyFlink Job](#running-the-pyflink-job)
```bash
$ docker-compose exec jobmanager ./bin/flink run -py /opt/pyflink-walkthrough/payment_msg_proccessing.py -d
```

Furthermore, you can also [create new kibana dashboards](https://www.elastic.co/guide/en/kibana/7.8/dashboard-create-new-dashboard.html) 
that visualize other aspects of the data in the Elasticsearch.
```